airflow/migrations/versions/0050_3_0_0_remove_pickled_data_from_dagrun_table.py

```diff
@@ -0,0 +1,118 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+remove pickled data from dagrun table.
+
+Revision ID: e39a26ac59f6
+Revises: eed27faa34e3
+Create Date: 2024-12-01 08:33:15.425141
+
+"""
+
+from __future__ import annotations
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy import text
+from sqlalchemy.dialects.mysql import LONGBLOB
+
+# revision identifiers, used by Alembic.
+revision = "e39a26ac59f6"
+down_revision = "eed27faa34e3"
+branch_labels = None
+depends_on = None
+airflow_version = "3.0.0"
+
+
+def upgrade():
+    """Apply remove pickled data from dagrun table."""
+    # Summary of the change:
+    # 1. Updating dag_run.conf column value to NULL.
+    # 2. Update the dag_run.conf column type to JSON from bytea
+
+    conn = op.get_bind()
+    dialect = conn.dialect.name
+
+    # Update the dag_run.conf column value to NULL
+    conn.execute(text("UPDATE dag_run set conf=null WHERE conf IS NOT NULL"))
```

Comment:
This just gets rid of all of the values? They may not all be pickled, right?

Reply:
I assumed that if we upgrade to this new version, all values in the configuration would be pickled. Should we add a condition to check for pickled data?

Reply:
cc: @kaxil

Reply:
Since DagRun.conf uses `PickleType`, all values should be pickled

https://github.com/apache/airflow/blob/b69441dc27888012f218683b09241e5db12c37a5/airflow/models/dagrun.py#L140